### ISSUE TITLE
LuigiClient.linkManager().navigate() doesn't send query parameters

### ISSUE THREAD

**Issue by DenisDuev** [SAP/luigi#2673]
**Description**

Using `LuigiClient.linkManager().navigate("/some/path?param=test)` with path that contains query parameters isn't working

Is this expected behavior or not? If this is the expected it would be great to have a note in the documentation that is not supported.

I have seen that in the Navigate with Intent it accepts "hash" so I assumed it also accepted query params `linkManager().navigate('/#?intent=Sales-settings?project=pr2&user=john')`

**Expected result**

navigate to new page with URL: `<host>/some/path?param=test`

**Actual result**

Query param is not send

 `<host>/some/path`

**Steps to reproduce**

call navigate with query param

**Troubleshooting**

I am thinking of using `addCoreSearchParams`

**Comment by ndricimrr** [comment]
Hi @DenisDuev 
Related to `LuigiClient.linkManager().navigate("/some/path?param=test)` :
In order to pass parameters through a **LuigiClient** navigation call, what you are looking for I think is the following [withParams](https://docs.luigi-project.io/docs/luigi-client-api/?section=examples-37) navigation option:
`LuigiClient.linkManager().withParams({param: test}).navigate('/some/path')` .
You can then receive the parameter on the other end with [getNodeParams](https://docs.luigi-project.io/docs/luigi-client-api/?section=getnodeparams) :
` LuigiClient.getNodeParams()` .
As for the confusion with the intent navigation, that is a different special navigation case that does not work like the generic navigation. Let us know if this answers your question.

**Comment by AlexandarNaydenov** [comment]
Hi @ndricimrr 
We decided to move to NodeParams as you suggested but it we have problems with the encoding.
We want to provide a URL as a node parameter so I have tried to do this:
```
LuigiClient.linkManager()
          .withoutSync()
          .fromVirtualTreeRoot()
          .withParams({url: sUrl})
          .navigate('');
```
To remove current path and add 'url' node parameter. 
For example the url could be: `https://raw.github.tools.sap/dxp/tech-docs-e2e-test-repository/main/component-with-mkdocs/README.md`
It is working fine but and add it as expected as: 
`?~url=https:%2F%2Fraw.github.tools.sap%2Fdxp%2Ftech-docs-e2e-test-repository%main%2Fcomponent-without-mkdocs%2FREADME.md`


But when we reload or open it in a new tab when we use `LuigiClient.getNodeParams()['url'];` to extract it it returns: 
`https:%26sol;%26sol;raw.github.tools.sap%26sol;dxp%26sol;tech-docs-e2e-test-repository%26sol;main%26sol;component-without-mkdocs%26sol;README.md`
Which looks like **HTML Entities** and I think it is not expected and it will be pain to decode back.

**Comment by ndricimrr** [comment]
@AlexandarNaydenov 
Hi there, the parameter value is intentionally encoded and needs to be decoded back, so this is expected - there should be a note in the documentation about this as well. 

However, your use case of passing a complete URL itself as a parameter value is quite interesting and something haven't seen before and I can see it might be somewhat of a pain here to decode.

In that case, maybe for your use case it would make more sense to use something else that might be more appropriate for this scenario when it comes to microfrontend navigation where you pass a URL as parameter value. 

One way that I am thinking could be helpful and also easier would be to use the [storageManager](https://docs.luigi-project.io/docs/luigi-client-api/?section=storagemanager) instead of passing by query/node parameters:
The following should work:
Set the item value before navigating:
`LuigiClient.storageManager().setItem('url','https://some-long-url').then(() => console.log('Value stored'));
`

and receive it on the other end:
`LuigiClient.storageManager().getItem('url').then((value) => console.log(value));`

The stored value then can be easily accessed and there's no need for encode/decoding. You can then clear the storage with the clear function in a similar fashion.

**Comment by AlexandarNaydenov** [comment]
@ndricimrr 
Hi, the storageManager will not work in my case because I wanted to change the url so it could be saved or sent to someone else and he arrived at the same page. And that can not be done with local storage.

Currently we are using **CoreSearchParams**, but I have understood that **NodeParams** is more the "Luigi way" for our case.
With `addCoreSearchParams and getCoreSearchParams` decoding is working perfectly and I would expect `addNodeParams and getNodeParams` to behave the same. Is it an expected behavior of nodeParams encoding to work differently from coreSearchParams`s one?

**Comment by ndricimrr** [comment]
Hi @AlexandarNaydenov , I see your point, I just checked the difference and it seems indeed that the addNodeParams function does not decode properly. I will investigate this further on, and let you know.

**Comment by AlexandarNaydenov** [comment]
> Hi @AlexandarNaydenov , I see your point, I just checked the difference and it seems indeed that the add(get)NodeParams function does not decode properly. I will investigate this further on, and let you know.

Thank you!

**Comment by ndricimrr** [comment]
Hi @AlexandarNaydenov, we are tracking this issue and noticed that it happens due to certain sanitising that we do internally. 

Will be issuing a fix to this issue soon.

In the meanwhile one small suggestion for this particular scenario could be the following: 

```
const encodedURL = encodeURIComponent('https://raw.github.tools.sap/dxp/tech-docs-e2e-test-repository/main/component-with-mkdocs/README.md');
LuigiClient.linkManager().withParams({param: encodedURL}).navigate('/some/path');

// and then receive it on the other end as
decodeURIComponent(LuigiClient.getNodeParams().param);
```

**Comment by AlexandarNaydenov** [comment]
> Hi @AlexandarNaydenov, we are tracking this issue and noticed that it happens due to certain sanitising that we do internally.
> 
> Will be issuing a fix to this issue soon.
> 
> In the meanwhile one small suggestion for this particular scenario could be the following:
> 
> ```
> const encodedURL = encodeURIComponent('https://raw.github.tools.sap/dxp/tech-docs-e2e-test-repository/main/component-with-mkdocs/README.md');
> LuigiClient.linkManager().withParams({param: encodedURL}).navigate('/some/path');
> 
> // and then receive it on the other end as
> decodeURIComponent(LuigiClient.getNodeParams().param);
> ```

Thank you, again! 
We decided to stick with CoreSearchParams for now and we will follow the progress of this issue. And when you are ready will we adapt to the NodeParams.
Have a nice day!

**Comment by ndricimrr** [comment]
Hi @AlexandarNaydenov , 
This issue was resolved in #2686 . It should be available on the next release/nightly.
